apps/ejabberd/src/mod_offline.erl

```diff
@@ -421,18 +416,21 @@ check_event_chatstates(From, To, Packet) ->
             false;
         %% There was an x:event element, and maybe also other stuff
         {El, _, _} when El /= false ->
-            case xml:get_subtag(El, <<"id">>) of
+            inspect_xevent(From, To, Packet, El)
+    end.
+
+inspect_xevent(From, To, Packet, XEvent) ->
+    case xml:get_subtag(XEvent, <<"id">>) of
```

Comment:
In new code we use `exml_query` instead of `xml`. Would it make sense to change this, since this line was refactored?